runtime/src/iree-amd-aie/aie_runtime/iree_aie_configure.cc

```diff
@@ -210,9 +210,12 @@ LogicalResult pushToBdQueueAndEnable(const AMDAIEDeviceModel &deviceModel,
 }
 
 LogicalResult addElfToTile(const AMDAIEDeviceModel &deviceModel,
-                           const TileLoc &tileLoc, const Path &elfPath,
-                           bool aieSim) {
+                           const TileLoc &tileLoc, Path &elfPath, bool aieSim) {
   auto devInst = const_cast<XAie_DevInst *>(&deviceModel.devInst);
+  if (!std::filesystem::exists(elfPath)) {
```

Comment:
`std::filesystem::exists` takes a const path (https://en.cppreference.com/w/cpp/filesystem/exists) so not sure why you de-consts this. But I like the error.

Reply:
It takes a const reference which means copy ellision can happen. I de-consted because prior I was doing `make_preferred` in here which mutates.

Reply:
restored